app/assets/javascripts/dashboard/views/dash/dataset_view.js

```diff
@@ -0,0 +1,159 @@
+// @flow
+/* eslint-disable jsx-a11y/href-no-hash */
+
+import _ from "lodash";
+import React from "react";
+import type { APIUserType, APIDatasetType } from "admin/api_flow_types";
+import Request from "libs/request";
+import Utils from "libs/utils";
+import moment from "moment";
+import { Spin, Input } from "antd";
+import SpotlightItemView from "./spotlight_item_view";
+import AdvancedDatasetView from "./advanced_dataset/advanced_dataset_view";
+
+const { Search } = Input;
+
+type Props = {
+  user: APIUserType,
+};
+
+function createThumbnailURL(datasetName, layers) {
+  const colorLayer = _.find(layers, { category: "color" });
+  if (colorLayer) {
+    return `/api/datasets/${datasetName}/layers/${colorLayer.name}/thumbnail`;
+  }
+  return "";
+}
+
+class DatasetView extends React.PureComponent {
+  props: Props;
+  state: {
+    currentDataViewType: "gallery" | "advanced",
+    datasets: Array<APIDatasetType>,
+    searchQuery: string,
+    isLoading: boolean,
+  } = {
+    currentDataViewType: "gallery",
+    datasets: [],
+    searchQuery: "",
+    isLoading: false,
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  async fetchData(): Promise<void> {
+    const url = "/api/datasets";
+    this.setState({ isLoading: true });
+    const datasets = await Request.receiveJSON(url);
+
+    const transformedDatasets = _.sortBy(
+      datasets.map(dataset => {
+        // since defaults doesn't override null...
```

Comment:
Yeah, there probably was a time in which backbone's (or lodash's) `defaultsTo` was used. Weirdly, it wasn't used in the backbone model, but I added it to the react version now, since I think it's correct (e.g., `baseDir` is sometimes not defined).